### PR TITLE
options/posix: implement getnameinfo()

### PR DESCRIPTION
**Pull request by Geertiebear**

### CODE REVIEW

File: options/posix/include/mlibc/lookup.hpp [hunk 1]
```diff
@@ -44,8 +46,10 @@ struct ai_buf {
 
 int lookup_name_dns(struct lookup_result &buf, const char *name,
 		frg::string<MemoryAllocator> &canon_name);
+int lookup_addr_dns(frg::span<char> name, frg::array<uint8_t, 16> &addr, int family);
```
Comment (avdgrinten) [inline options/posix/include/mlibc/lookup.hpp:49]: `string_view` probably makes more sense than `span<char>`.

Comment (Geertiebear) [reply options/posix/include/mlibc/lookup.hpp:49]: A ``string_view`` is immutable, but I need to be able to overwrite the buffer

Comment (avdgrinten) [reply options/posix/include/mlibc/lookup.hpp:49]: Alright. So the result is fixed-length? If it isn't, it might make more sense to return a `string`.

Comment (Geertiebear) [reply options/posix/include/mlibc/lookup.hpp:49]: Yeah, it's always limited by the size of the buffer the user passes.

**Comment by avdgrinten** [comment]
From my side, we can merge this PR. Is it properly tested and does it work?

**Comment by Geertiebear** [comment]
Yeah, Dennis said it worked, so it should be good to go.

**Comment by avdgrinten** [comment]
Alright, let's merge this. Thanks for the PR!

**Pull request merged**